nifi-nar-bundles/nifi-websocket-bundle/nifi-websocket-services-jetty/src/main/java/org/apache/nifi/websocket/jetty/JettyWebSocketClient.java

```diff
@@ -229,6 +260,11 @@ protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
     public void startClient(final ConfigurationContext context) throws Exception {
         configurationContext = context;
 
+        connectCount = configurationContext.getProperty(CONNECTION_ATTEMPT_COUNT).evaluateAttributeExpressions().asInteger();
+        maximumBackoffMillis = configurationContext.getProperty(MAXIMUM_BACKOFF_TIME).asTimePeriod(TimeUnit.MILLISECONDS);
+        initialBackoffMillis = configurationContext.getProperty(INITIAL_BACKOFF_TIME).asTimePeriod(TimeUnit.MILLISECONDS);
+        backoffJitterMillis = (long) (initialBackoffMillis * BACKOFF_JITTER_FACTOR * ThreadLocalRandom.current().nextDouble(-1, 1));
```

Comment:
Generally we want to avoid using ThreadLocal as a mechanism in NiFi though in this case it might be fine.  That said what is the need for a random number generated here?  Can we just toss out this random and have an explicit value?

Reply:
Since it is only used at the initialization I can change it to Random to adhere to the guidelines.

There's a maintaining thread that periodically checks and re-establishes connections for active sessions. When there are numerous active sessions, this could strain the server. By introducing randomness or jitter to the retry intervals, clients are less likely to retry simultaneously, spreading out the load more evenly and reducing the chances of congestion.